pkg/payload/error.go

```diff
@@ -0,0 +1,91 @@
+package payload
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// Error is a wrapper for errors that occur during a payload sync.
+type Error struct {
+	Nested  error
+	Reason  string
+	Message string
+	Name    string
+
+	Task *Task
+}
+
+func (e *Error) Error() string {
+	return e.Message
+}
+
+func (e *Error) Cause() error {
+	return e.Nested
+}
+
+func (e *Error) Summary() string {
```

Comment:
This was separate because it was a specific domain error being applied - the error message for API short status is not the same as the message in other contexts.  Summary isn’t a property of the error, it is an interpretation of it.

So would prefer not to do this part.